src/wp-includes/script-loader.php

```diff
@@ -1843,6 +1843,38 @@ function wp_just_in_time_script_localization() {
 	);
 }
 
+/**
+ * Prints a loader script if there is text/template registered script.
+ *
+ * @since 6.3.0
+ *
+ * When added to the DOM, this script converts any text/template scripts
+ * associated with a handle into type/javascript, and executes them.
+ */
+function wp_print_delayed_inline_script_loader() {
+	$wp_scripts = wp_scripts();
+
+	if ( $wp_scripts->has_delayed_inline_script() ) {
+		$output    = <<<JS
+function wpLoadAfterScripts( handle ) {
+	var scripts, newScript, i, len;
+	scripts = document.querySelectorAll(
+		'[type="text/template"][data-wp-executes-after="' + handle + '"]'
+	);
+	for ( i = 0, len = scripts.length; i < len; i++ ) {
+		newScript = scripts[ i ].cloneNode( true );
+		newScript.type = "text/javascript";
+		scripts[ i ].parentNode.replaceChild( newScript, scripts[ i ] );
+	}
+}
```

Comment:
Here's a patched version to account for the CSP `nonce`:

```js
(function () {
  // Capture the nonce of the currentScript so we can use it when evaluating after inline scripts.
  var nonce = document.currentScript.nonce;

  window.wpLoadAfterScripts = function wpLoadAfterScripts(handle) {
    var scripts, newScript, i, len;
    scripts = document.querySelectorAll(
      '[type="text/template"][data-wp-executes-after="' + handle + '"]'
    );
    for (i = 0, len = scripts.length; i < len; i++) {
      if (nonce && nonce !== scripts[i].nonce) {
        console.error(
          "CSP nonce check failed for after inline script. Execution aborted.",
          scripts[i]
        );
        continue;
      }
      newScript = scripts[i].cloneNode(true);
      newScript.type = "text/javascript";
      scripts[i].parentNode.replaceChild(newScript, scripts[i]);
    }
  };
})();
```

Demo: https://defer-scripts-with-after-scripts.glitch.me/